src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/mappers/EfilingFilingPackageMapper.java

```diff
@@ -30,15 +28,52 @@
     @Mapping(target = "ldcxCourtDivisionCd", source = "filingPackage.court.division")
     EfilingFilingPackage toEfilingFilingPackage(Submission submission);
 
-    @Mapping(target = "amendsAnotherDocumentYn", source = "isAmendment")
-    @Mapping(target = "clientFileNameTxt", source = "name")
-    @Mapping(target = "documentDescriptionTxt", source = "description")
-    @Mapping(target = "documentTypeCd", source = "type")
-    EfilingDocument toEfilingDocument(Document document);
+    @Mapping(target = "amendsAnotherDocumentYn", source = "document.isAmendment", defaultValue = "false")
+    @Mapping(target = "clientFileNameTxt", source = "document.name")
+    @Mapping(target = "documentDescriptionTxt", source = "document.description")
+    @Mapping(target = "documentSubtypeCd", constant = "ODOC" )
+    @Mapping(target = "documentTypeCd", source = "document.type")
+    @Mapping(target = "entUserId", source = "submission.clientId")
+    @Mapping(target = "entDtm",  expression = "java(ca.bc.gov.open.jag.efilingcommons.utils.DateUtils.getCurrentXmlDate())")
+    //TODO this is the constructed file name
+    @Mapping(target = "filePath", source = "document.name")
+    //TODO this will be pulled from the config
+    @Mapping(target = "fileServer", source = "serverName")
+    @Mapping(target = "uploadStateCd", constant = SubmissionConstants.SUBMISSION_UPLOAD_STATE_CD)
+    @Mapping(target = "milestones", source = "milestones")
+    @Mapping(target = "payments", source = "payments")
+    @Mapping(target = "statuses", source = "statuses")
+    @Mapping(target = "packageSeqNo", source = "index")
+    EfilingDocument toEfilingDocument(Document document, Submission submission, Integer index,
+                                      String serverName, List<EfilingDocumentMilestone> milestones,
+                                      List<EfilingDocumentPayment> payments, List<EfilingDocumentStatus> statuses);
+
+    @Mapping(target = "entUserId", source = "submission.clientId")
+    @Mapping(target = "entDtm",  expression = "java(ca.bc.gov.open.jag.efilingcommons.utils.DateUtils.getCurrentXmlDate())")
+    @Mapping(target = "milestoneDtm",  expression = "java(ca.bc.gov.open.jag.efilingcommons.utils.DateUtils.getCurrentXmlDate())")
+    @Mapping(target = "milestoneTypeCd",  constant = "ASUB")
```

Comment:
Would you mind extracting all the constants to keys? So we have quick access to all those values?

Reply:
This one is just there till we get the way to set the value from Jaime. But we can make a it a constant